client/model/src/main/java/io/smallrye/graphql/client/model/TypeModels.java

```diff
@@ -0,0 +1,22 @@
+package io.smallrye.graphql.client.model;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This is just a first step into having a more complete model of the GraphQL types in the client.
+ * It currently only holds the subtypes of interfaces that are used to model unions.
+ * In the long run, this should contain a complete model of all Java types relevant for the typesafe
+ * GraphQL client.
+ */
+public class TypeModels {
```

Comment:
I don't fully understand the purpose of this class, can't this data be clumped together inside https://github.com/smallrye/smallrye-graphql/blob/ceb755444bbc4993d3436e2897103d28ecc2a513/client/model/src/main/java/io/smallrye/graphql/client/model/ClientModel.java which is registered a CDI bean (wrapped in https://github.com/smallrye/smallrye-graphql/blob/ceb755444bbc4993d3436e2897103d28ecc2a513/client/model/src/main/java/io/smallrye/graphql/client/model/ClientModels.java)? 
That would also avoid having to use static fields, which is a major problem for wildfly

Reply:
Or is keeping reference to implementations necessary at all? I guess it could just be reflected in the query string that is pre-generated for each graphql operation

Reply:
I like that idea: we could produce all query strings at startup also for the reflection-based approach. Nice! I'm not sure how much effort that would be. I had thought that adding support for Unions would be a quick thing, but it doesn't look like it will be 😁

About the `static`: I had forgotten. Do you have a suggestion how to fix it? Should we use the `CDI` class to find an application-scoped instance of the `TypeModels`, of if we'd put it into `ClientModels`, make that application-scoped?!?

Reply:
Wouldn't it be enough to annotate the class as `@ApplicationScoped` and add a `beans.xml`? In the `TypeInfo` we could then use the `CDI` class.

Reply:
That would mean the CDI container would create an instance, but we currently build an instance manually at build-time, and then run it through a bytecode recorder and register the instance as a synthetic bean. Potentially we could rework it to a regular bean created by CDI and somehow set the value during startup, or something... but why wouldn't you be able to use the `CDI` class this way? It's still a bean in the CDI container, that will work normally.

Reply:
The suggestion I just postet in the PR would make this unnecessary.